internal/model/model.go

```diff
@@ -84,8 +84,12 @@ type service interface {
 }
 
 type Model struct {
-	cfg             *config.Wrapper
-	db              *leveldb.DB
+	cfg *config.Wrapper
+	db  *leveldb.DB
+
+	cache    *db.NamespacedKV
+	cacheMut sync.Mutex
```

Comment:
The thing with the locks here is to make sure the cached values are consistent with the localVersion number they are saved for. The individual key/values are safe enough on their own. An RWMutex would work, but I'm not sure there's any real gain; reading the few values is so fast I don't think there's any contention.